packages/animations/lib/src/fade_transition.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/utils/curves.dart';
+import 'package:flutter/material.dart';
+
+/// Displays a modal above the current contents of the app.
+///
+/// This function displays the [FadeModalRoute], which transitions in
+/// with the Material fade transition.
+///
+/// Content below the modal is dimmed with a [ModalBarrier].
+///
+/// ```dart
+/// /// Sample widget that uses [showModalWithFadeTransition].
+/// class MyHomePage extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       body: Center(
+///         child: RaisedButton(
+///           onPressed: () {
+///             showModalWithFadeTransition(
+///               context: context,
+///               child: FlutterLogoModal(),
+///             );
+///           },
+///           child: Icon(Icons.add),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+///
+/// /// Displays a modal with the FlutterLogo on it.
+/// class FlutterLogoModal extends StatelessWidget {
+///   const _FlutterLogoModal();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return Column(
+///       mainAxisAlignment: MainAxisAlignment.center,
+///       children: <Widget>[
+///         Center(
+///           child: ConstrainedBox(
+///             constraints: const BoxConstraints(
+///               maxHeight: 300,
+///               maxWidth: 300,
+///               minHeight: 250,
+///               minWidth: 250,
+///             ),
+///             child: const Material(
+///               child: Center(child: FlutterLogo(size: 250)),
+///             ),
+///           ),
+///         ),
+///       ],
+///     );
+///   }
+/// }
+/// ```
+///
+/// The `context` argument is used to look up the [Navigator] for the
+/// modal. It is only used when the method is called. Its corresponding widget
+/// can be safely removed from the tree before the modal is closed.
+///
+/// The `useRootNavigator` argument is used to determine whether to push the
+/// modal to the [Navigator] furthest from or nearest to the given `context`.
+/// By default, `useRootNavigator` is `true` and the modal route created by
+/// this method is pushed to the root navigator.
+///
+/// If the application has multiple [Navigator] objects, it may be necessary to
```

Comment:
nit: it seems like this paragraph should just be a continuation of the previous one?